### DISCUSSION TITLE
How to request support for a new lens?

### DISCUSSION THREAD

**Discussion by stercomp** [lensfun/lensfun#1781] [Q&A] [answered]
In the old sourceforge site there is mention of making some raw images available but can't find any clear documentation here on the process of requesting support for a new lens. Do I need to clone this repo and use the Django tool?

P.S. specifically I want to request support for Viltrox 33mm F1.4 for Fuji-X

**Comment by stercomp** [comment, marked as answer]
Ahh.. found it:
https://pixls.us/articles/create-lens-calibration-data-for-lensfun/
It's a lot of work lol. Hoping somebody will get to it before me but else I'll get to it later this year.

**Reply by tuxfanx** [reply]
You don't have to do the calibration yourself. You can [upload the corresponding raw images here](https://wilson.bronger.org/calibration). For a prime lens, one or two pictures are sufficient for distortion and tca calibration. A little more effort would be needed for the vignetting correction.

**Reply by smartintec** [reply]
I want to contribute but I'm a bit uncertain about what kind of pictures to upload, as the example shows architecture with straight lines.
Furthermore the only link I found to a hugin tutorial seems to point to a hacked blog website.

As I just setup the Ad*be calibration "Checkerboard" with proper lighting and my new cameras on a geared head, I want to take some useful pictures for this project.
So I can send RAWs in various focal length and apertures with a plain black/white checkerboard.
I use full manual, low ISO, and for now f8.0 shooting in different focal length with zoom lenses.

What is best for this purpose here:
1) One pic checkerboard filling the frame or 
2) Nine pictures with that checkerboard in all quadrants like for the LCP tool?

**Reply by smartintec** [reply]
Sorry, I think I got it:
There have to be continuous straight lines from one border of the frame to the other.
They should be at least 5 meters away from the lens.
So no checkerboard. I need something big - like architecture ;)

**Reply by tuxfanx** [reply]
Here you will find some hints:
[calibration with lens_calibrate.py and taking pictures](https://pixls.us/articles/create-lens-calibration-data-for-lensfun/)
[calibration and taking pictures](https://wilson.bronger.org/calibration)
[video 1 calibration with Hugin](https://kameratrollet.se/create-your-own-lens-correction-data-for-lensfun/)
[video 2 calibration with Hugin ](https://www.youtube.com/watch?v=_p0fR-nH6cU)
[lensfuns coverage](https://wilson.bronger.org/lensfun_coverage.html)
[lensfuns database](https://github.com/lensfun/lensfun/tree/master/data/db)
[calibration with calibrate.py](https://wilson.bronger.org/lens_calibration_tutorial/)

**Reply by smartintec** [reply]
Thank you for helpful and quick reply.
I already read some of them but currently have no working python environment.
The LCP conversion tool also looks promising, as I already created LCP profiles that work fine in RawTherapee.
Now looking how this all works with zoom lenses until I got python up and running.
Look up table or interpolation.. but that would be offtopic here.
